servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1076,24 +1077,25 @@ public boolean rename(long fileId, TachyonURI dstPath)
       // Get the inodes of the src and dst parents.
       Inode srcParentInode = mInodeTree.getInodeById(srcInode.getParentId());
       if (!srcParentInode.isDirectory()) {
-        return false;
+        throw new UnsupportedOperationException("File to rename must have a valid parent.");
       }
       Inode dstParentInode = mInodeTree.getInodeByPath(dstParentURI);
       if (!dstParentInode.isDirectory()) {
-        return false;
+        throw new UnsupportedOperationException("Destination must have a valid parent.");
       }
 
       // Make sure destination path does not exist
       InodeDirectory dstParentDirectory = (InodeDirectory) dstParentInode;
       String[] dstComponents = PathUtils.getPathComponents(dstPath.getPath());
       if (dstParentDirectory.getChild(dstComponents[dstComponents.length - 1]) != null) {
-        return false;
+        throw new FileAlreadyExistsException(
+            ExceptionMessage.FILE_ALREADY_EXISTS.getMessage(dstPath));
       }
 
       // Now we remove srcInode from it's parent and insert it into dstPath's parent
       long opTimeMs = System.currentTimeMillis();
       if (!renameInternal(fileId, dstPath, false, opTimeMs)) {
-        return false;
+        throw new IOException("Internal error.");
```

Comment:
should renameInternal throw exceptions instead of returning a boolean to provide for more informative error message?